adaptors/library/jobs/satusehat-create-observations.js

```diff
@@ -0,0 +1,140 @@
+// Create an observation in Satusehat
+
+fn(state => {
+  function createObservationData(item, state, result) {
+    const patientReference = `Patient/${state.patient.ihs_number}`;
+    const encounterReference = `Encounter/${state.encounterId}`;
+    const encounterDisplay = `${state.patient.full_name}'s ${state.visit.properties.visit_type} on ${state.visit.properties.visit_date}`;
+    return {
+      resourceType: 'Observation',
+      status: 'final',
+      category: [
+        {
+          coding: item.category,
+        },
+      ],
+      code: {
+        coding: item.code,
+      },
+      subject: {
+        reference: patientReference,
+      },
+      performer: [
+        {
+          reference: `Practitioner/N10000001`,
+          display: 'Voigt',
+        },
+      ],
+      encounter: {
+        reference: encounterReference,
+        display: encounterDisplay,
+      },
+      effectiveDateTime: state.visit.visit_date,
+      ...result,
+    };
+  }
+
+  /**
+   * Filters and maps observations based on the visit.
+   */
+  function mapObservations(state) {
+    return state.visit.observations.map(item => {
+      const { value_type: valueType, value } = item;
+      const dataObjects = [];
+
+      if (valueType === 'valueCodeableConcept') {
+        value.forEach(value => {
+          const result = {
+            [valueType]: {
+              coding: [JSON.parse(value.snomed_code)],
```

Comment:
Curious as to why we have all this JSON parsing? Is that really needed? Can we cheat in this example and pretend the input is well formatted?

Reply:
I think we need to show them how we create the different mappings since it is an important aspect when creating an obervation. cc @AishaHassen

Reply:
@josephjclark  I have removed the parsing